core/src/main/java/org/apache/calcite/adapter/enumerable/EnumerableTraitsUtils.java

```diff
@@ -148,4 +149,53 @@ private static boolean isCollationOnTrivialExpr(
       return null;
     }
   }
+
+  // This function can be reused when a Join's traits pass-down shall only
```

Comment:
minor detail: I think it would be better to use javadoc comment here (instead of line comment). Same for `deriveTraitsForJoin`.

Reply:
Updated to javadoc. Thanks for you suggestion!